### PR TITLE
read efpt.config.json from all diretories

### PR DESCRIPTION
**Pull request by pedromcrocha**

### CODE REVIEW

File: src/GUI/EFCorePowerTools.Shared/Models/ConfigModel.cs [hunk 1]
```diff
@@ -23,13 +24,32 @@ public string ConfigPath
             }
         }
 
+        /// <summary>
+        /// Gets or sets the project path to the eftp.*.config.json file.
```
Comment (ErikEJ) [inline src/GUI/EFCorePowerTools.Shared/Models/ConfigModel.cs:28]: Just: Gets or sets the project path.

File: src/GUI/EFCorePowerTools.Shared/Models/ConfigModel.cs [hunk 2]
```diff
@@ -23,13 +24,32 @@ public string ConfigPath
...
+
+
         public string DisplayName
         {
             get
             {
                 if (!string.IsNullOrEmpty(_configPath))
                 {
-                    return Path.GetFileName(_configPath);
+                    //I dont test all use of this class, that is why I test if projectpath is used
```
Comment (ErikEJ) [inline src/GUI/EFCorePowerTools.Shared/Models/ConfigModel.cs:48]: If it is the root config file, return: "efpt.config.json"
For other files, return the full path below the project path.

So the complete UI would look like this:

```
efpt.config.json
Folder1\efpt.config.extra.json
Folder2\efpt.config.extra.json

```

File: src/GUI/EFCorePowerTools.Shared/Models/ConfigModel.cs [hunk 3]
```diff
@@ -23,13 +24,32 @@ public string ConfigPath
...
+        }
+
+
         public string DisplayName
         {
             get
             {
                 if (!string.IsNullOrEmpty(_configPath))
                 {
-                    return Path.GetFileName(_configPath);
```
Comment (ErikEJ) [inline src/GUI/EFCorePowerTools.Shared/Models/ConfigModel.cs:32]: Remove the comment

File: src/GUI/EFCorePowerTools/Extensions/ProjectExtensions.cs [hunk 4]
```diff
@@ -66,17 +66,20 @@ public static List<string> GetConfigFiles(this Project project)
                 return result;
             }
 
-            var file = Directory.GetFiles(projectPath, "efpt.config.json");
-            result.AddRange(file);
+            //var file = Directory.GetFiles(projectPath, "efpt.config.json");
```
Comment (ErikEJ) [inline src/GUI/EFCorePowerTools/Extensions/ProjectExtensions.cs:69]: I think you need to leave this in, if there are any special config files and the standard one, the standard one will be missed if you do not include these 2 lines

File: src/GUI/EFCorePowerTools/Extensions/ProjectExtensions.cs [hunk 5]
```diff
@@ -66,17 +66,20 @@ public static List<string> GetConfigFiles(this Project project)
...
+            //result.AddRange(file);
 
-            var files = Directory.GetFiles(projectPath, "efpt.*.config.json");
+            //var files = Directory.GetFiles(projectPath, "efpt.*.config.json");
```
Comment (ErikEJ) [inline src/GUI/EFCorePowerTools/Extensions/ProjectExtensions.cs:72]: Do check commented code in

**Review by ErikEJ — CHANGES_REQUESTED** [review]

**Comment by ErikEJ** [comment]
Could you share a screenshot?

File: src/GUI/EFCorePowerTools/Extensions/ProjectExtensions.cs (hunk 4 above)
Comment (pedromcrocha) [reply src/GUI/EFCorePowerTools/Extensions/ProjectExtensions.cs:69]: no need.
 "efpt.*config.json" will get 
efpt.config.json
efpt.sample.config.json
efpt.sample2config.json

alway get the standart one

Comment (ErikEJ) [reply src/GUI/EFCorePowerTools/Extensions/ProjectExtensions.cs:69]: Yes, but this method must return at least on  entry, even if no file exists! Look at the logic in the caller.

Comment (pedromcrocha) [reply src/GUI/EFCorePowerTools/Extensions/ProjectExtensions.cs:69]: and will, the rest of the code ensures that

            if (result.Count() == 0)
            {
                result.Add(Path.Combine(projectPath, "efpt.config.json"));
            }

Comment (ErikEJ) [reply src/GUI/EFCorePowerTools/Extensions/ProjectExtensions.cs:69]: OK! 😅

**Comment by pedromcrocha** [comment]
[cid:image001.png@01D70243.7E26B7D0]

[cid:image002.png@01D70243.7E26B7D0]

**Comment by ErikEJ** [comment]
I think you need to upload images via web.

**Comment by pedromcrocha** [comment]
image

**Comment by ErikEJ** [comment]
Looks great, but you must rebase on latest master.

**Comment by ErikEJ** [comment]
Thanks for  a great improvement

**Pull request merged**